src/metpy/xarray.py

```diff
@@ -1000,11 +1000,17 @@ def quantify(self):
         by this operation. Do not utilize on moderate- to large-sized remote datasets before
         subsetting!
         """
-        return self._dataset.map(lambda da: da.metpy.quantify())
+        _attr_save = self._dataset.attrs.copy()
+        dataset = self._dataset.map(lambda da: da.metpy.quantify())
+        dataset.attrs = _attr_save
+        return dataset
```

Comment:
Unless `self._dataset` is modified in place, shouldn't this do it?
```suggestion
        dataset = self._dataset.map(lambda da: da.metpy.quantify())
        dataset.attrs = self._dataset.attrs.copy()
        return dataset
```

Reply:
And same below in `dequantify`?

Reply:
Could go even further?
```suggestion
        return self._dataset.map(lambda da: da.metpy.quantify()).assign_attrs(self._dataset.attrs)
```